### ISSUE TITLE
@OnetoMany / @ManyToOne fields in an @Superclass annotated interface

### ISSUE THREAD

**Issue by polmabri** [requery/requery#321]
Hey,

is it possible to use @OnetoMany / @ManyToOne fields in a @Superclass annotated interface?

Kotlin example:
```
@Entity
interface X {
    
    @get:Key
    @get:Generated
    var id: Int

    var name: String

    @get:ManyToOne
    var a: A
}

@Superclass
interface A {

    @get:OneToMany
    var xes: Set<X>
}

@Entity
interface B : A {
    @get:Key
    @get:Generated
    var id: Int
}

@Entity
interface C : A {
    @get:Key
    @get:Generated
    var id: Int
}
```

The expected schema:

![unbenannt](https://cloud.githubusercontent.com/assets/874881/19966102/96726182-a1ca-11e6-8e33-9d5f171bb1b8.PNG)

**Comment by npurushe** [comment]
No currently you can't have an relational mapping to a superclass type. However I think having a superclass type refer to other non superclass types should work fine. Currently you can solve this by moving the relational references out of the superclass types (e.g. define X_B and X_C). However something like this may be possible in a later version.

**Comment by npurushe** [comment]
Should be resolved with the changes from #461, will be in the next update
